### PR TITLE
Document how to configure JSON from an extension

### LINKED ISSUE #4339: Document how an extension can customize JSON handling
Follow up of [this](https://github.com/quarkusio/quarkus/pull/4212#pullrequestreview-296700733) discussion.

### PR DESCRIPTION
**Pull request by loicmathieu**
This fixes #4339

Ping @geoand

### CODE REVIEW

File: docs/src/main/asciidoc/extension-authors-guide.adoc [hunk 1]
```diff
@@ -1233,6 +1233,130 @@ HealthBuildItem addHealthCheck(AgroalBuildTimeConfig agroalBuildTimeConfig) {
 }
 ----
 
+=== Customizing Json handling from an extension
+It is often needed for an extension to register serializer/deserializer for specific types that the extension provides.
```
Comment (geoand) [inline docs/src/main/asciidoc/extension-authors-guide.adoc]: I would rephrase like so:

```
Extensions often need to register serializers and/or deserializers for types the extension provides.
```

File: docs/src/main/asciidoc/extension-authors-guide.adoc [hunk 2]
```diff
@@ -1233,6 +1233,130 @@ HealthBuildItem addHealthCheck(AgroalBuildTimeConfig agroalBuildTimeConfig) {
 }
 ----
 
+=== Customizing Json handling from an extension
```
Comment (geoand) [inline docs/src/main/asciidoc/extension-authors-guide.adoc]: Let's use `JSON` as is done in the `rest-json` guide.

File: docs/src/main/asciidoc/extension-authors-guide.adoc [hunk 3]
```diff
@@ -1233,6 +1233,130 @@ HealthBuildItem addHealthCheck(AgroalBuildTimeConfig agroalBuildTimeConfig) {
...
+It is often needed for an extension to register serializer/deserializer for specific types that the extension provides.
+
+For this, both JsonB and Jackson extension provide a way to register serializer/deserializer from within an
```
Comment (geoand) [inline docs/src/main/asciidoc/extension-authors-guide.adoc]: `JsonB` should `JSON-B` everywhere.

File: docs/src/main/asciidoc/extension-authors-guide.adoc [hunk 4]
```diff
@@ -1233,6 +1233,130 @@ HealthBuildItem addHealthCheck(AgroalBuildTimeConfig agroalBuildTimeConfig) {
...
+extension deployment module.
+
+Keep in minds that not everybody will need json, so you need to make it optional.
```
Comment (geoand) [inline docs/src/main/asciidoc/extension-authors-guide.adoc]: `minds` -> `mind`

File: docs/src/main/asciidoc/extension-authors-guide.adoc [hunk 5]
```diff
@@ -1233,6 +1233,130 @@ HealthBuildItem addHealthCheck(AgroalBuildTimeConfig agroalBuildTimeConfig) {
...
+It is strongly advised to provide customization for both JsonB and Jackson.
```
Comment (geoand) [inline docs/src/main/asciidoc/extension-authors-guide.adoc]: I would add a qualifier before this, saying: `If an extension intends to provide JSON related customization, ....`

File: docs/src/main/asciidoc/extension-authors-guide.adoc [hunk 6]
```diff
@@ -1233,6 +1233,130 @@ HealthBuildItem addHealthCheck(AgroalBuildTimeConfig agroalBuildTimeConfig) {
...
+[source%nowrap,xml]
+----
+<dependency>
+  <groupId>io.quarkus</groupId>
+  <artifactId>quarkus-jsonb</artifactId>
+  <optional>true</optional>
+</dependency>
+----
+
+Then create a serializer or a deserializer for JsonB, example is from the `mongodb-panache` extension.
```
Comment (geoand) [inline docs/src/main/asciidoc/extension-authors-guide.adoc]: `or` -> `and/or`

Comment (geoand) [reply docs/src/main/asciidoc/extension-authors-guide.adoc]: `example is from the` -> `an example of which can be seen in the`

**Review by geoand — COMMENTED** [review]
Looks great! I just added some small comments

File: docs/src/main/asciidoc/extension-authors-guide.adoc (hunk 1 above)
Comment (loicmathieu) [reply docs/src/main/asciidoc/extension-authors-guide.adoc]: done

File: docs/src/main/asciidoc/extension-authors-guide.adoc (hunk 2 above)
Comment (loicmathieu) [reply docs/src/main/asciidoc/extension-authors-guide.adoc]: done

File: docs/src/main/asciidoc/extension-authors-guide.adoc (hunk 3 above)
Comment (loicmathieu) [reply docs/src/main/asciidoc/extension-authors-guide.adoc]: done

File: docs/src/main/asciidoc/extension-authors-guide.adoc (hunk 4 above)
Comment (loicmathieu) [reply docs/src/main/asciidoc/extension-authors-guide.adoc]: done

File: docs/src/main/asciidoc/extension-authors-guide.adoc (hunk 5 above)
Comment (loicmathieu) [reply docs/src/main/asciidoc/extension-authors-guide.adoc]: done

File: docs/src/main/asciidoc/extension-authors-guide.adoc (hunk 6 above)
Comment (loicmathieu) [reply docs/src/main/asciidoc/extension-authors-guide.adoc]: done

**Comment by loicmathieu** [comment]
@geoand comments addressed

**Review by geoand — APPROVED** [review]
Thanks for this @loicmathieu!
